frontend/src/scenes/debug/DebugScene.tsx

```diff
@@ -33,13 +47,32 @@ function QueryDebug({ query, setQuery, queryKey }: QueryDebugProps): JSX.Element
                     setQuery={(query) => setQuery(JSON.stringify(query, null, 2))}
                 />
             ) : (
-                <Query
-                    query={query}
-                    setQuery={(query) => setQuery(JSON.stringify(query, null, 2))}
-                    context={{
-                        showQueryEditor: true,
-                    }}
-                />
+                <div className="space-y-4">
+                    <QueryEditor query={query} setQuery={setQuery} />
+                    <Modifiers
+                        setQuery={
+                            parsed?.source
+                                ? (query) => setQuery(JSON.stringify({ ...parsed, source: query }, null, 2))
+                                : (query) => setQuery(JSON.stringify(query, null, 2))
+                        }
+                        query={parsed?.source ?? parsed}
+                        response={response}
+                    />
+                    <LemonDivider />
+                    <Query
+                        uniqueKey={queryKey}
+                        query={query}
+                        setQuery={(query) => setQuery(JSON.stringify(query, null, 2))}
+                    />
+                    {response && parsed && (isDataTableNode(parsed as Node) || isInsightVizNode(parsed as Node)) ? (
+                        <CodeEditor
```

Comment:
Would it make sense to pass readonly={true}? Though looking at other usages, we don't do that anywhere else, so 🤷

Reply:
Perhaps... but that'll require another 2h of CI massaging... so nope 😅